albogdan/skule-vote
# Solve ballot algorithm edge case

## Overview

- Solves an edge case where:
   - Number of candidate (including ron) = number of seats for that election
   - Every non-ron passes the quota and wins a round
   - Ron does not meet the quota to win the final round/seat
- The issue (carried over from the original ballot.js) is that in the last round, because the last candidate (ron) does not pass the quota, it called `backwardsEliminationProcess` expecting to delete the candidate with the lowest votes. RON is the only candidate, thus has the lowest number of votes, but can never be considered for elimination. There is no if statement to check if `eliminationList` is empty, thus we get an index error when trying to determine who to eliminate.

## Unit Tests Created

- Creates the edge case explained above


## Steps to QA

- Run the tests, make sure they all pass
- Delete the if statement I added in `ballot.py`. Run the tests again. `ballot.py` should throw `IndexError: list index out of range` 
- Repeat for `ballot_new.py`. You should get a `ValueError: min() arg is an empty sequence`
